src/viking.js

```diff
@@ -1,11 +1,101 @@
 // Soldier
-class Soldier {}
+class Soldier {
+    constructor(health, strength) {
+        this.health = health;
+        this.strength = strength;
+    };
+
+    attack() {
+        return this.strength;
+    };
+    receiveDamage(damage){
+        this.health -= damage
+    }
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+    constructor(name, health,strength) {
+        super(health, strength);
+        this.name = name;
+    }
+
+    receiveDamage(damage){
+        this.health -= damage;
+
+        if (this.health > 0){
+            return `${this.name} has received ${damage} points of damage`;
+        } else {
+            return `${this.name} has died in act of combat`;
+        }
+    }
+
+    battleCry(){
+        return "Odin Owns You All!";
+    }
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+    constructor(health, strength) {
+        super(health, strength);
+    }
+
+    receiveDamage(damage){
+        this.health -= damage;
+
+        if (this.health > 0){
+            return `A Saxon has received ${damage} points of damage`;
+        } else {
+            return `A Saxon has died in combat`;
+        }
+    }
+}
 
 // War
-class War {}
+class War {
+    constructor(){
+        this.vikingArmy = [];
+        this.saxonArmy = [];
+
+    }
+    addViking(vikingObject){
+        this.vikingArmy.push(vikingObject);
+
+    };
+
+    addSaxon(saxonObject){
+        this.saxonArmy.push(saxonObject);
+
+    };
+    vikingAttack() {
+        let randomSaxon = Math.floor(Math.random()*this.saxonArmy.length);
+        let randomViking = Math.floor(Math.random()*this.vikingArmy.length);
+        let result = this.saxonArmy[randomSaxon].receiveDamage(this.vikingArmy[randomViking].strength)
+
+        if (this.saxonArmy[randomSaxon].health <= 0) {
+            this.saxonArmy.pop(randomSaxon);
```

Comment:
This `.pop()` method, always removes the last index from the array. Even you passing the random saxon index inside, it will always remove the last index. You are passing the jasmine tests, but it is not correct, as you are always removing the last saxon from the `saxonArmy` array.